### DISCUSSION TITLE
General questions about GLSP

### DISCUSSION THREAD

**Discussion by tmirza** [eclipse-glsp/glsp#1443] [General]
Hi All,

I have some general questions about Eclipse GLSP as I am evaluating it as an option to migrate our existing modeling editors in different products(one using Capella and the other one custom). Based on reading the docs, I am considering either going with Sirius Web or Eclipse GLSP. Eclipse GLSP provides more flexibility and by using it, I can develop a common frontend (unified look and feel) for users and I can still keep model sources as they are in current products. I will have to develop a server framework in such a way that it can operate on two different model sources based on the product being used. So overall, migration will be easy. What do you think about it?

Also, looking at the GitHub project, I have the following questions:

1. The last release of the Eclipse GLSP was in July 2023. Is there any release planned? I am trying to understand if project is still in active releases.
2. Is there any way to know how many companies/products have adopted Eclipse GLSP so far? Is there any adopters list?
3. Is this project still in active development? I only see one open pull request and it is also open for quite some time.

Thanks,
Taimoor

**Comment by planger** [comment]
Hi Taimoor,  

Thank you for reaching out and for your interest in Eclipse GLSP! It’s great to hear that you’re evaluating it as an option for migrating your modeling editors. Your idea of using Eclipse GLSP to achieve a unified look and feel with full control over the UX and the tool integration, while keeping the model sources of your existing products sounds like a solid approach. GLSP’s flexibility is one of its strengths, and supporting multiple source models is a common use case. How a GLSP server handles source models—be it files, databases, REST APIs, or something else—is entirely up to the server implementation, making it highly adaptable to your needs.  

To address your specific questions:  

### **1. Project Activity and Releases** 

The latest release of Eclipse GLSP was on **July 23, 2024**, and the project follows a release cycle of approximately every three months, aligned with the Eclipse Theia Community releases. So, yes, the project is very much alive and actively maintained by both EclipseSource and the broader community. If you look beyond the umbrella repository (https://github.com/eclipse-glsp/glsp), you’ll see the actual activity in the child repositories, such as:

- [GLSP Client](https://github.com/eclipse-glsp/glsp-client)
- [GLSP Server](https://github.com/eclipse-glsp/glsp-server)
- [GLSP Server Node](https://github.com/eclipse-glsp/glsp-server-node)

These repos are where the bulk of ongoing development happens.  

### **2. Adopters**

You can find a list of public adopters on the [Eclipse Cloud DevTools Adopters page](https://ecdtools.eclipse.org/adopters/) and a gallery of use cases on the [GLSP website](https://eclipse.dev/glsp/gallery/). While these give you a sense of the adoption, there are several other organizations that use GLSP but have chosen not to be listed publicly. Judging by community engagement and what we see in conference talks, there are likely much more adopters out there.

### **3. Community and Engagement** 

The Eclipse GLSP community is quite active, with ongoing discussions happening in the [GitHub Discussions section](https://github.com/eclipse-glsp/glsp/discussions). If you have questions or need guidance, it’s a great place to connect with both the maintainers and the broader community.  

Feel free to share further details or ask if you have more specific questions!
